win_installer.py

```diff
@@ -152,23 +165,36 @@ def pypi_wheels_in(requirements):
         print("-", requirement, end=" ")
         package = yarg.get(name)
         releases = package.release(version)
-        if any(r.package_type == "wheel" for r in releases):
+        if not releases:
+            feedback = (
+                "WARNING: Just installed but not found at PyPI."
+                " (bad meta-data?)"
+            )
```

Comment:
So in this case we are only writing a warning to the console.

The reason I would prefer to fail the build is because I honestly don't think we will ever catch a warning like this unless there is a problem with the build.

Now, if PyNsist fails somewhere else then there is no problem.

However, if we ended up downloading the package source code and compiling in the build server, the build could still pass, right? Or does PyNsist refuse to build anything from source?
If the build passes in this case then we could end up with a version of Mu that works in most of our tests system, but fails in some user environments and we might not catch that.

Generally if we compile something we should aim to do that in the lowest common denominator (Win 7?), as we had issues related to that in the past, specifically with some dynamically linked win libraries that are common, but not everywhere.

Reply:
> So in this case we are only writing a warning to the console.
> 
> The reason I would prefer to fail the build is because I honestly don't think we will ever catch a warning like this unless there is a problem with the build.
> 
> Now, if PyNsist fails somewhere else then there is no problem.

That is the case as of today, with regards to `PyQtChart`, yes.

> However, if we ended up downloading the package source code and compiling in the build server, the build could still pass, right? Or does PyNsist refuse to build anything from source?

AFAICT, `pynsist` does not build anything: it collects wheel dependencies from PyPI, on one hand, and copies any other dependencies from the Python environment it is running on -- and fails miserably at handling `PyQtChart`, for example, but does its thing nicely with the `tkinter` or `turtle` modules.

> If the build passes in this case then we could end up with a version of Mu that works in most of our tests system, but fails in some user environments and we might not catch that.
> 
> Generally if we compile something we should aim to do that in the lowest common denominator (Win 7?), as we had issues related to that in the past, specifically with some dynamically linked win libraries that are common, but not everywhere.

Agreed, even though `pynsist` itself does not build anything, in the more strict Python packaging build sense. Additionally, and into the realms of #893, if we're looking for consistency and reproducibility, we should probably lock all direct and indirect dependencies in `setup.py`. Such a policy would have avoided the macOS failing builds we're having now due to the `python-dateutil` version conflict that pip doesn't know how to solve. :)